// ng-zorro-antd/esm5/menu/nz-menu.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { merge, BehaviorSubject, Subject } from 'rxjs';
var NzMenuService = /** @class */ (function () {
    function NzMenuService() {
        this.menuItemClick$ = new Subject();
        this.theme$ = new Subject();
        this.mode$ = new BehaviorSubject('vertical');
        this.inlineIndent$ = new BehaviorSubject(24);
        this.check$ = merge(this.theme$, this.mode$, this.inlineIndent$);
        this.theme = 'light';
        this.mode = 'vertical';
        this.inlineIndent = 24;
        this.menuOpen$ = new BehaviorSubject(false);
    }
    /**
     * @param {?} menu
     * @return {?}
     */
    NzMenuService.prototype.onMenuItemClick = /**
     * @param {?} menu
     * @return {?}
     */
    function (menu) {
        this.menuItemClick$.next(menu);
    };
    /**
     * @param {?} mode
     * @return {?}
     */
    NzMenuService.prototype.setMode = /**
     * @param {?} mode
     * @return {?}
     */
    function (mode) {
        this.mode = mode;
        this.mode$.next(mode);
    };
    /**
     * @param {?} theme
     * @return {?}
     */
    NzMenuService.prototype.setTheme = /**
     * @param {?} theme
     * @return {?}
     */
    function (theme) {
        this.theme = theme;
        this.theme$.next(theme);
    };
    /**
     * @param {?} indent
     * @return {?}
     */
    NzMenuService.prototype.setInlineIndent = /**
     * @param {?} indent
     * @return {?}
     */
    function (indent) {
        this.inlineIndent = indent;
        this.inlineIndent$.next(indent);
    };
    NzMenuService.decorators = [
        { type: Injectable }
    ];
    return NzMenuService;
}());
export { NzMenuService };
if (false) {
    /** @type {?} */
    NzMenuService.prototype.isInDropDown;
    /** @type {?} */
    NzMenuService.prototype.menuItemClick$;
    /** @type {?} */
    NzMenuService.prototype.theme$;
    /** @type {?} */
    NzMenuService.prototype.mode$;
    /** @type {?} */
    NzMenuService.prototype.inlineIndent$;
    /** @type {?} */
    NzMenuService.prototype.check$;
    /** @type {?} */
    NzMenuService.prototype.theme;
    /** @type {?} */
    NzMenuService.prototype.mode;
    /** @type {?} */
    NzMenuService.prototype.inlineIndent;
    /** @type {?} */
    NzMenuService.prototype.menuOpen$;
}
