// ng-zorro-antd/esm5/page-header/nz-page-header-cells.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive } from '@angular/core';
var NzPageHeaderTitleDirective = /** @class */ (function () {
    function NzPageHeaderTitleDirective() {
    }
    NzPageHeaderTitleDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-title, [nz-page-header-title]',
                    host: {
                        class: 'ant-page-header-title-view-title'
                    }
                },] }
    ];
    return NzPageHeaderTitleDirective;
}());
export { NzPageHeaderTitleDirective };
var NzPageHeaderSubtitleDirective = /** @class */ (function () {
    function NzPageHeaderSubtitleDirective() {
    }
    NzPageHeaderSubtitleDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-subtitle, [nz-page-header-subtitle]',
                    host: {
                        class: 'ant-page-header-title-view-sub-title'
                    }
                },] }
    ];
    return NzPageHeaderSubtitleDirective;
}());
export { NzPageHeaderSubtitleDirective };
var NzPageHeaderContentDirective = /** @class */ (function () {
    function NzPageHeaderContentDirective() {
    }
    NzPageHeaderContentDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-content, [nz-page-header-content]',
                    host: {
                        class: 'ant-page-header-content-view'
                    }
                },] }
    ];
    return NzPageHeaderContentDirective;
}());
export { NzPageHeaderContentDirective };
var NzPageHeaderTagDirective = /** @class */ (function () {
    function NzPageHeaderTagDirective() {
    }
    NzPageHeaderTagDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-tags, [nz-page-header-tags]',
                    host: {
                        class: 'ant-page-header-title-view-tags'
                    }
                },] }
    ];
    return NzPageHeaderTagDirective;
}());
export { NzPageHeaderTagDirective };
var NzPageHeaderExtraDirective = /** @class */ (function () {
    function NzPageHeaderExtraDirective() {
    }
    NzPageHeaderExtraDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-extra, [nz-page-header-extra]',
                    host: {
                        class: 'ant-page-header-title-view-extra'
                    }
                },] }
    ];
    return NzPageHeaderExtraDirective;
}());
export { NzPageHeaderExtraDirective };
var NzPageHeaderFooterDirective = /** @class */ (function () {
    function NzPageHeaderFooterDirective() {
    }
    NzPageHeaderFooterDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-page-header-footer, [nz-page-header-footer]',
                    host: {
                        class: 'ant-page-header-footer'
                    }
                },] }
    ];
    return NzPageHeaderFooterDirective;
}());
export { NzPageHeaderFooterDirective };
var NzPageHeaderBreadcrumbDirective = /** @class */ (function () {
    function NzPageHeaderBreadcrumbDirective() {
    }
    NzPageHeaderBreadcrumbDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'nz-breadcrumb[nz-page-header-breadcrumb]'
                },] }
    ];
    return NzPageHeaderBreadcrumbDirective;
}());
export { NzPageHeaderBreadcrumbDirective };
