// ng-zorro-antd/esm2015/menu/nz-menu.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { merge, BehaviorSubject, Subject } from 'rxjs';
export class NzMenuService {
    constructor() {
        this.menuItemClick$ = new Subject();
        this.theme$ = new Subject();
        this.mode$ = new BehaviorSubject('vertical');
        this.inlineIndent$ = new BehaviorSubject(24);
        this.check$ = merge(this.theme$, this.mode$, this.inlineIndent$);
        this.theme = 'light';
        this.mode = 'vertical';
        this.inlineIndent = 24;
        this.menuOpen$ = new BehaviorSubject(false);
    }
    /**
     * @param {?} menu
     * @return {?}
     */
    onMenuItemClick(menu) {
        this.menuItemClick$.next(menu);
    }
    /**
     * @param {?} mode
     * @return {?}
     */
    setMode(mode) {
        this.mode = mode;
        this.mode$.next(mode);
    }
    /**
     * @param {?} theme
     * @return {?}
     */
    setTheme(theme) {
        this.theme = theme;
        this.theme$.next(theme);
    }
    /**
     * @param {?} indent
     * @return {?}
     */
    setInlineIndent(indent) {
        this.inlineIndent = indent;
        this.inlineIndent$.next(indent);
    }
}
NzMenuService.decorators = [
    { type: Injectable }
];
if (false) {
    /** @type {?} */
    NzMenuService.prototype.isInDropDown;
    /** @type {?} */
    NzMenuService.prototype.menuItemClick$;
    /** @type {?} */
    NzMenuService.prototype.theme$;
    /** @type {?} */
    NzMenuService.prototype.mode$;
    /** @type {?} */
    NzMenuService.prototype.inlineIndent$;
    /** @type {?} */
    NzMenuService.prototype.check$;
    /** @type {?} */
    NzMenuService.prototype.theme;
    /** @type {?} */
    NzMenuService.prototype.mode;
    /** @type {?} */
    NzMenuService.prototype.inlineIndent;
    /** @type {?} */
    NzMenuService.prototype.menuOpen$;
}
