// ng-zorro-antd/esm5/ng-zorro-antd.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NzAffixModule } from './affix/nz-affix.module';
import { NzAlertModule } from './alert/nz-alert.module';
import { NzAnchorModule } from './anchor/nz-anchor.module';
import { NzAutocompleteModule } from './auto-complete/nz-autocomplete.module';
import { NzAvatarModule } from './avatar/nz-avatar.module';
import { NzBackTopModule } from './back-top/nz-back-top.module';
import { NzBadgeModule } from './badge/nz-badge.module';
import { NzBreadCrumbModule } from './breadcrumb/nz-breadcrumb.module';
import { NzButtonModule } from './button/nz-button.module';
import { NzCalendarModule } from './calendar/nz-calendar.module';
import { NzCardModule } from './card/nz-card.module';
import { NzCarouselModule } from './carousel/nz-carousel.module';
import { NzCascaderModule } from './cascader/nz-cascader.module';
import { NzCheckboxModule } from './checkbox/nz-checkbox.module';
import { NzCollapseModule } from './collapse/nz-collapse.module';
import { NzCommentModule } from './comment/nz-comment.module';
import { NzNoAnimationModule } from './core/no-animation/nz-no-animation.module';
import { NzWaveModule } from './core/wave/nz-wave.module';
import { NzDatePickerModule } from './date-picker/date-picker.module';
import { NzDividerModule } from './divider/nz-divider.module';
import { NzDrawerModule } from './drawer/nz-drawer.module';
import { NzDropDownModule } from './dropdown/nz-dropdown.module';
import { NzEmptyModule } from './empty/nz-empty.module';
import { NzFormModule } from './form/nz-form.module';
import { NzGridModule } from './grid/nz-grid.module';
import { NzI18nModule } from './i18n/nz-i18n.module';
import { NzIconModule } from './icon/nz-icon.module';
import { NzInputNumberModule } from './input-number/nz-input-number.module';
import { NzInputModule } from './input/nz-input.module';
import { NzLayoutModule } from './layout/nz-layout.module';
import { NzListModule } from './list/nz-list.module';
import { NzMentionModule } from './mention/nz-mention.module';
import { NzMenuModule } from './menu/nz-menu.module';
import { NzMessageModule } from './message/nz-message.module';
import { NzModalModule } from './modal/nz-modal.module';
import { NzNotificationModule } from './notification/nz-notification.module';
import { NzPageHeaderModule } from './page-header/nz-page-header.module';
import { NzPaginationModule } from './pagination/nz-pagination.module';
import { NzPopconfirmModule } from './popconfirm/nz-popconfirm.module';
import { NzPopoverModule } from './popover/nz-popover.module';
import { NzProgressModule } from './progress/nz-progress.module';
import { NzRadioModule } from './radio/nz-radio.module';
import { NzRateModule } from './rate/nz-rate.module';
import { NzSelectModule } from './select/nz-select.module';
import { NzSkeletonModule } from './skeleton/nz-skeleton.module';
import { NzSliderModule } from './slider/nz-slider.module';
import { NzSpinModule } from './spin/nz-spin.module';
import { NzStatisticModule } from './statistic/nz-statistic.module';
import { NzStepsModule } from './steps/nz-steps.module';
import { NzSwitchModule } from './switch/nz-switch.module';
import { NzTableModule } from './table/nz-table.module';
import { NzTabsModule } from './tabs/nz-tabs.module';
import { NzTagModule } from './tag/nz-tag.module';
import { NzTimePickerModule } from './time-picker/nz-time-picker.module';
import { NzTimelineModule } from './timeline/nz-timeline.module';
import { NzToolTipModule } from './tooltip/nz-tooltip.module';
import { NzTransferModule } from './transfer/nz-transfer.module';
import { NzTreeSelectModule } from './tree-select/nz-tree-select.module';
import { NzTreeModule } from './tree/nz-tree.module';
import { NzUploadModule } from './upload/nz-upload.module';
export { NzAffixComponent, NzAffixModule } from './affix';
export { NzAlertComponent, NzAlertModule } from './alert';
export { NzAnchorLinkComponent, NzAnchorComponent, NzAnchorModule } from './anchor';
export { NzAvatarComponent, NzAvatarModule } from './avatar';
export { NzBackTopComponent, NzBackTopModule } from './back-top';
export { NzBadgeComponent, NzBadgeModule } from './badge';
export { NzBreadCrumbItemComponent, NZ_ROUTE_DATA_BREADCRUMB, NzBreadCrumbComponent, NzBreadCrumbModule } from './breadcrumb';
export { NzButtonComponent, NzButtonGroupComponent, NzButtonModule } from './button';
export { NzCalendarComponent, NzCalendarModule } from './calendar';
export { NzCardGridDirective, NzCardComponent, NzCardModule, NzCardLoadingComponent, NzCardMetaComponent, NzCardTabComponent } from './card';
export { NzCarouselModule, NzCarouselComponent, NzCarouselContentDirective } from './carousel';
export { NzCheckboxComponent, NzCheckboxModule, NzCheckboxGroupComponent, NzCheckboxWrapperComponent } from './checkbox';
export { NzCollapsePanelComponent, NzCollapseComponent, NzCollapseModule } from './collapse';
export { NzCommentModule, NzCommentComponent, NzCommentAvatarDirective, NzCommentContentDirective, NzCommentActionHostDirective, NzCommentActionComponent } from './comment';
export { CandyDate, NzDatePickerModule, NzDatePickerComponent, NzRangePickerComponent, NzMonthPickerComponent, NzWeekPickerComponent, NzYearPickerComponent } from './date-picker';
export { NzDividerComponent, NzDividerModule } from './divider';
export { DRAWER_ANIMATE_DURATION, NzDrawerComponent, NzDrawerModule, DrawerBuilderForService, NzDrawerService, NzDrawerRef } from './drawer';
export { NzDropdownContextComponent, NzDropDownComponent, NzDropDownDirective, NzDropdownService, NzDropDownButtonComponent, NzDropDownModule } from './dropdown';
export {} from './drawer';
export { NzEmbedEmptyComponent, NzEmptyComponent, NzEmptyModule, NzEmptyService, NZ_DEFAULT_EMPTY_CONTENT, NZ_EMPTY_COMPONENT_NAME, emptyImage, simpleEmptyImage } from './empty';
export { NzFormModule, NzFormDirective, NzFormControlComponent, NzFormExplainComponent, NzFormItemComponent, NzFormExtraComponent, NzFormLabelComponent, NzFormSplitComponent, NzFormTextComponent } from './form';
export { Breakpoint, NzRowDirective, NzColDirective, NzGridModule } from './grid';
export { NzI18nModule, NzI18nService, NZ_DATE_CONFIG, DateHelperService, ar_EG, bg_BG, ca_ES, cs_CZ, da_DK, de_DE, el_GR, en_GB, en_US, es_ES, et_EE, fa_IR, fi_FI, fr_BE, fr_FR, is_IS, it_IT, ja_JP, ko_KR, nb_NO, nl_BE, nl_NL, pl_PL, pt_BR, pt_PT, ru_RU, sk_SK, sl_SI, sr_RS, sv_SE, th_TH, tr_TR, uk_UA, vi_VN, zh_CN, zh_TW, NZ_I18N, NZ_DATE_LOCALE } from './i18n';
export { NzIconModule, NzIconDirective, NZ_ICONS, NZ_ICON_DEFAULT_TWOTONE_COLOR, DEFAULT_TWOTONE_COLOR, NZ_ICONS_USED_BY_ZORRO, NzIconService } from './icon';
export { NzInputGroupComponent, NzInputModule, NzInputDirective } from './input';
export { NzInputNumberComponent, NzInputNumberModule } from './input-number';
export { NzContentComponent, NzFooterComponent, NzHeaderComponent, NzLayoutComponent, NzSiderComponent, NzLayoutModule } from './layout';
export { NzListItemMetaComponent, NzListItemComponent, NzListComponent, NzListModule } from './list';
export { NzMentionModule, NzMentionComponent, NZ_MENTION_TRIGGER_ACCESSOR, NzMentionTriggerDirective, NzMentionSuggestionDirective } from './mention';
export { NzMenuFactory, NzMenuDirective, NzMenuGroupComponent, NzMenuDividerDirective, NzMenuItemDirective, NzSubMenuComponent, NzMenuModule } from './menu';
export { NzPageHeaderModule, NzPageHeaderComponent, NzPageHeaderTitleDirective, NzPageHeaderSubtitleDirective, NzPageHeaderContentDirective, NzPageHeaderTagDirective, NzPageHeaderExtraDirective, NzPageHeaderFooterDirective, NzPageHeaderBreadcrumbDirective } from './page-header';
export { NzPaginationComponent, NzPaginationModule } from './pagination';
export { NzProgressModule, NzProgressComponent } from './progress';
export { NzRadioButtonComponent, NzRadioGroupComponent, NzRadioComponent, NzRadioModule } from './radio';
export { NzRateComponent, NzRateModule } from './rate';
export { NzOptionGroupComponent, NzOptionContainerComponent, NzOptionComponent, NzSelectComponent, NzSelectModule } from './select';
export { NzSpinComponent, NzSpinModule } from './spin';
export { NzCountdownComponent, NzStatisticComponent, NzStatisticModule } from './statistic';
export { NzStepsComponent, NzStepComponent, NzStepsModule } from './steps';
export { NzSwitchComponent, NzSwitchModule } from './switch';
export { NzTableComponent, NzTableModule, NzTbodyDirective, NzTdComponent, NzThComponent, NzTheadComponent, NzTrDirective } from './table';
export { NzTabBodyComponent, NzTabLabelDirective, NzTabComponent, NzTabsInkBarDirective, NzTabsModule, NzTabsNavComponent, NzTabChangeEvent, NzTabSetComponent } from './tabs';
export { NzTimelineItemComponent, NzTimelineComponent, NzTimelineModule } from './timeline';
export { NzTransferListComponent, NzTransferSearchComponent, NzTransferComponent, NzTransferModule } from './transfer';
export { NzUploadBtnComponent, NzUploadListComponent, NzUploadComponent, NzUploadModule } from './upload';
export { NzTagComponent, NzTagModule } from './tag';
export { NzAutocompleteModule, NzAutocompleteComponent, getNzAutocompleteMissingPanelError, NZ_AUTOCOMPLETE_VALUE_ACCESSOR, NzAutocompleteTriggerDirective, NzOptionSelectionChange, NzAutocompleteOptionComponent, NzAutocompleteOptgroupComponent } from './auto-complete';
export { NzMessageBaseService, NzMessageService, NzMessageModule, NzMessageComponent, NzMessageContainerComponent, NZ_MESSAGE_DEFAULT_CONFIG, NZ_MESSAGE_CONFIG, NZ_MESSAGE_DEFAULT_CONFIG_PROVIDER } from './message';
export { NzTimePickerComponent, NzTimePickerModule } from './time-picker';
export { NzToolTipComponent, NzTooltipDirective, NzToolTipModule } from './tooltip';
export { NzSkeletonComponent, NzSkeletonModule } from './skeleton';
export { NzSliderComponent, NzSliderModule, NzSliderHandleComponent, NzSliderMarksComponent, NzSliderStepComponent, NzSliderTrackComponent, isValueARange, isConfigAObject, Marks } from './slider';
export { NzPopoverComponent, NzPopoverDirective, NzPopoverModule } from './popover';
export { NZ_NOTIFICATION_DEFAULT_CONFIG, NZ_NOTIFICATION_CONFIG, NZ_NOTIFICATION_DEFAULT_CONFIG_PROVIDER, NzNotificationComponent, NzNotificationModule, NzNotificationService, NzNotificationContainerComponent } from './notification';
export { NzPopconfirmComponent, NzPopconfirmDirective, NzPopconfirmModule } from './popconfirm';
export { NzModalComponent, NzModalRef, NzModalModule, NzModalService, NZ_MODAL_CONFIG } from './modal';
export { isShowSearchObject, isChildOption, isParentOption, NzCascaderComponent, NzCascaderModule, NzCascaderService } from './cascader';
export { NzTreeModule, NzTreeServiceFactory, NzTreeComponent, NzTreeNodeComponent, NzTreeNode, NzTreeBaseService, NzTreeService, isCheckDisabled, isInArray } from './tree';
export { NzTreeSelectComponent, NzTreeSelectModule } from './tree-select';
export {} from './time-picker';
export { VERSION } from './version';
export { NzWaveRenderer, NZ_WAVE_GLOBAL_CONFIG_FACTORY, NZ_WAVE_GLOBAL_DEFAULT_CONFIG, NZ_WAVE_GLOBAL_CONFIG, NzWaveDirective, NzWaveModule } from './core/wave';
export { isNotNil, isNil, shallowEqual, isInteger, isEmpty, filterNotEmptyNode, isNonEmptyString, isTemplateRef, isComponent, toBoolean, toNumber, toCssPixel, valueFunctionProp, InputBoolean, InputCssPixel, InputNumber, getRegExp, getMentions, padStart, padEnd, getRepeatedElement, getCaretCoordinates, createDebugEle, properties, throttleByAnimationFrameDecorator, timeUnits } from './core/util';
export { NzNoAnimationModule, NzNoAnimationDirective } from './core/no-animation';
var NgZorroAntdModule = /** @class */ (function () {
    function NgZorroAntdModule() {
    }
    /**
     * @deprecated Use `NgZorroAntdModule` instead.
     */
    /**
     * @deprecated Use `NgZorroAntdModule` instead.
     * @return {?}
     */
    NgZorroAntdModule.forRoot = /**
     * @deprecated Use `NgZorroAntdModule` instead.
     * @return {?}
     */
    function () {
        return {
            ngModule: NgZorroAntdModule
        };
    };
    NgZorroAntdModule.decorators = [
        { type: NgModule, args: [{
                    exports: [
                        NzButtonModule,
                        NzCalendarModule,
                        NzGridModule,
                        NzSwitchModule,
                        NzSelectModule,
                        NzMenuModule,
                        NzMentionModule,
                        NzAnchorModule,
                        NzAffixModule,
                        NzDropDownModule,
                        NzLayoutModule,
                        NzBreadCrumbModule,
                        NzPaginationModule,
                        NzStepsModule,
                        NzInputModule,
                        NzCheckboxModule,
                        NzInputNumberModule,
                        NzSliderModule,
                        NzRateModule,
                        NzBadgeModule,
                        NzRadioModule,
                        NzAlertModule,
                        NzSpinModule,
                        NzPageHeaderModule,
                        NzProgressModule,
                        NzTabsModule,
                        NzIconModule,
                        NzCardModule,
                        NzAvatarModule,
                        NzTimelineModule,
                        NzTransferModule,
                        NzCarouselModule,
                        NzCollapseModule,
                        NzCommentModule,
                        NzTableModule,
                        NzDatePickerModule,
                        NzDividerModule,
                        NzDrawerModule,
                        NzFormModule,
                        NzListModule,
                        NzI18nModule,
                        NzUploadModule,
                        NzAutocompleteModule,
                        NzTagModule,
                        NzMessageModule,
                        NzNotificationModule,
                        NzPopoverModule,
                        NzToolTipModule,
                        NzPopconfirmModule,
                        NzModalModule,
                        NzBackTopModule,
                        NzCascaderModule,
                        NzTreeModule,
                        NzTreeSelectModule,
                        NzTimePickerModule,
                        NzWaveModule,
                        NzNoAnimationModule,
                        NzSkeletonModule,
                        NzStatisticModule,
                        NzEmptyModule
                    ]
                },] }
    ];
    return NgZorroAntdModule;
}());
export { NgZorroAntdModule };
